tests/publiccloud/cloud_netconfig.pm

```diff
@@ -50,10 +50,25 @@ sub run {
       $instance->ssh_script_output(cmd => $grep_cloudvm_ipv4_cmd);
     chomp($local_eth0_ip);
 
-    my $query_meta_ipv4_cmd =
-      'curl -H Metadata:true '
-      . '"http://169.254.169.254/metadata/instance/network/interface/0/ipv4/'
-      . 'ipAddress/0/privateIpAddress?api-version=2023-07-01&format=text"';
+    my $query_meta_ipv4_cmd = "";
+    if (check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE')) {
```

Comment:
in `publiccloud::utils` you may find functions `is_ec2` , `is_gce` and `is_azure` ( see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/utils.pm#L36 ) please use them instead

Reply:
I have implemented your suggestion.